src/main/java/org/osiam/storage/query/ExtensionQueryField.java

```diff
@@ -48,7 +49,11 @@ public Predicate addFilter(Root<UserEntity> root, FilterConstraint constraint,
         if (constraint != FilterConstraint.PRESENT && (field.getType() == ExtensionFieldType.INTEGER ||
                 field.getType() == ExtensionFieldType.DECIMAL)) {
 
-            value = numberPadder.pad(value);
+            try {
+                value = numberPadder.pad(value);
+            } catch (IllegalArgumentException ex) {
+                throw new InvalidFilterException(ex.getMessage());
```

Comment:
I dont' think we are doing anything with the cause further up the stack, so I don't really think keeping the cause wins us anything. Do you see it differently?